File: backend/src/main/java/com/documed/backend/visits/Visit.java
package com.documed.backend.visits;

import com.documed.backend.attachments.Attachment;
import com.documed.backend.notifications.Notification;
import com.documed.backend.prescription.Prescription;
import com.documed.backend.referrals.Referral;
import com.documed.backend.schedules.TimeSlot;
import com.documed.backend.services.Service;
import com.documed.backend.users.User;
import java.math.BigDecimal;
import java.util.List;
import lombok.Data;
import lombok.NonNull;

@Data
public class Visit {
  private final int id;
  @NonNull private VisitStatus status;
  private String interview;
  private String diagnosis;
  private String recommendations;
  private BigDecimal totalCost;
  @NonNull private Facility facility;
  @NonNull private Service service;
  private String patientInformation;
  @NonNull private User patient;
  @NonNull private User doctor;
  private Feedback feedback;
  private Attachment attachment;
  private List<Prescription> prescriptions;
  private List<Referral> referrals;
  private List<Notification> notifications;
  @NonNull private List<TimeSlot> timeSlots;
}

File: frontend/src/shared/types/Visit.ts
import { AttachmentLite } from './Attachment';
import { VisitStatus } from './enums';
import { FacilityLite } from './Facility';
import { FeedbackLite } from './Feedback';
import { NotificationLite } from './Notification';
import { PrescriptionLite } from './Prescription';
import { ReferralLite } from './Referral';
import { ServiceLite } from './Service';
import { TimeSlot } from './TimeSlot';
import { UserLite } from './User';

export interface Visit {
  facility: FacilityLite;
  service: ServiceLite;
  patient: UserLite;
  doctor: UserLite;
  feedback: FeedbackLite | null;
  attachments: AttachmentLite[] | null;
  prescriptions: PrescriptionLite[] | null;
  referrals: ReferralLite[] | null;
  notifications: NotificationLite[] | null;
}

export interface VisitLite {
  id: number;
  status: VisitStatus;
  interview: string | null;
  diagnosis: string | null;
  recommendations: string | null;
  totalCost: number | null;
  timeSlots: TimeSlot[];
  patientInformation: string | null;
}
